### PR TITLE
DT-2929: ✨ Use email address from results

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/nomis/repository/StaffRepository.kt [hunk 1]
```diff
@@ -1,6 +0,0 @@
-package uk.gov.justice.digital.hmpps.oauth2server.nomis.repository
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/nomis/repository/StaffRepository.kt:1]: not used

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/nomis/service/NomisUserApiService.kt [hunk 2]
```diff
@@ -72,23 +73,21 @@ class NomisUserApiService(
         .build()
     }
       .retrieve()
-      .bodyToMono(NomisUserSummaryList::class.java)
+      .bodyToMono(object : ParameterizedTypeReference<List<NomisUserSummaryDto>>() {})
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/nomis/service/NomisUserApiService.kt:76]: seems a bit neater than defining a class unnecessarily

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/NomisUserService.kt [hunk 3]
```diff
@@ -31,16 +31,20 @@ abstract class NomisUserService(
   fun getNomisUsersByEmail(email: String): List<NomisUserPersonDetails> {
     val emailLowered = email.lowercase()
 
+    // Find all users in auth with the specified email address
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/NomisUserService.kt:34]: added bit of comments to aid understanding

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.kt [hunk 4]
```diff
@@ -151,40 +150,23 @@ class UserService(
...
     val authUsersByUsername = authUserService
-      .findAuthUsersByUsernames(nomisUsernames)
-      .filter {
-        !it.email.isNullOrBlank() && it.source == nomis
-      }.map { it.username to it }
-      .toMap()
-
-    val missingUsernames = nomisUsernames.minus(authUsersByUsername.keys)
-
-    val emailsByUsername = verifyEmailService.getExistingEmailAddressesForUsernames(missingUsernames)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.kt:165]: don't need separate call for this as first call returns email addresses too

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**